### ISSUE TITLE
remove non-ascii characters

### ISSUE THREAD

**Issue by andreas-junghanns** [modelica/fmi-standard#688]
We found strange double quotes and ->; add commit hook to check for non-ASCII

**Comment by t-sommer** [comment]
I wrote a simple linter that checks for non-ascii characters and whitespaces at the end of lines. Here's the result:

```
Problems found in ./docs/3_2_model_exchange_api.adoc

line 74: non-ascii characters found:
A value of `+1` indicates that latexmath:[z_i] is increasing, while a value of `−1` indicates a decreasing latexmath:[z_i].
                                                                                ^                                           

Problems found in ./docs/6___appendix.adoc

line 11: non-ascii characters found:
### A.1 Version 1.0 – FMI for Model Exchange
                    ^                        

line 19: non-ascii characters found:
- Hilding Elmqvist, Dassault Systèmes, Sweden
                                 ^            

line 22: non-ascii characters found:
- Hans Olsson, Dassault Systèmes, Sweden
                            ^            

line 28: non-ascii characters found:
- Dymola by Peter Nilsson, Dan Henriksson, Carl Fredrik Abelson, and Sven Erik Mattson, Dassault Systèmes,
                                                                                                     ^     

line 40: non-ascii characters found:
- Hilding Elmqvist, Dassault Systèmes, Lund, Sweden
                                 ^                  

line 44: non-ascii characters found:
- Carsten Kübler, TWT, Germany
           ^                   

line 49: non-ascii characters found:
- Peter Nilsson, Dassault Systèmes, Lund, Sweden
                              ^                  

line 50: non-ascii characters found:
- Hans Olsson, Dassault Systèmes, Lund, Sweden
                            ^                  

line 53: non-ascii characters found:
- Daniel Weil, Dassault Systèmes, Grenoble, France
                            ^                      

line 61: non-ascii characters found:
### A.2 Version 1.0 – FMI for Co-Simulation
                    ^                       

line 65: non-ascii characters found:
FMI for Co-Simulation was developed in three subgroups: "Solver Coupling" headed by Martin Arnold (University Halle) and Torsten Blochwitz (ITI), "Tool Coupling" headed by Jörg-Volker Peetz (Fraunhofer SCAI), and "Control Logic" headed by Manuel Monteiro (Atego).
                                                                                                                                                                             ^                                                                                          

line 71: non-ascii characters found:
- Christoph Clauß, Fraunhofer IIS EAS, Dresden, Germany
                ^                                       

line 74: non-ascii characters found:
- Jörg-Volker Peetz, Fraunhofer SCAI, St. Augustin, Germany
   ^                                                        

line 80: non-ascii characters found:
- Master algorithms by Christoph Clauß (Fraunhofer IIS EAS)
                                     ^                      

line 88: non-ascii characters found:
- Christoph Clauß, Fraunhofer IIS EAS, Dresden, Germany
                ^                                       

line 91: non-ascii characters found:
- Dietmar Neumerkel, Daimler AG, Böblingen, Germany
                                  ^                 

line 93: non-ascii characters found:
- Jörg-Volker Peetz, Fraunhofer SCAI, St. Augustin, Germany
   ^                                                        

line 97: non-ascii characters found:
### A.3 Version 2.0 – FMI for Model Exchange and Co-Simulation
                    ^                                          

line 269: non-ascii characters found:
- Hilding Elmqvist, Dassault Systèmes, Sweden
                                 ^            

line 270: non-ascii characters found:
- Rüdiger Franke, ABB AG, Germany
   ^                              

line 277: non-ascii characters found:
- Hans Olsson, Dassault Systèmes, Sweden
                            ^            

line 286: non-ascii characters found:
- Dymola by Peter Nilsson, Karl Wernersson, and Sven Erik Mattson, Dassault Systèmes, Sweden
                                                                                ^            

line 293: non-ascii characters found:
- xMOD by Mongi ben Gaid, Bertrand Hugon, Bruno Léty, and Fabien Debertolis, IFPEN, France
                                                 ^                                         

line 297: non-ascii characters found:
The open source FMITest library (https://svn.fmi-standard.org/fmi/branches/public/Test_FMUs/_FMIModelicaTest/FMITest/help/FMITest.html) to test difficult cases of connected FMUs was implemented by Martin Otter (DLR, Germany) based on suggestions by Hilding Elmqvist (Dassault Systèmes, Sweden) and Torsten Blochwitz (ITI, Germany).
                                                                                                                                                                                                                                                                                        ^                                                   

line 310: non-ascii characters found:
- Hilding Elmqvist, Dassault Systèmes, Sweden
                                 ^            

line 311: non-ascii characters found:
- Rüdiger Franke, ABB AG, Germany
   ^                              

line 318: non-ascii characters found:
- Karsten Krügel, dSPACE GmbH, Germany
            ^                          

line 323: non-ascii characters found:
- Sven Erik Mattsson, Dassault Systèmes, Sweden
                                   ^            

line 325: non-ascii characters found:
- Monika Mühlbauer, Siemens AG, Germany
          ^                             

line 327: non-ascii characters found:
- Peter Nilsson, Dassault Systèmes, Sweden
                              ^            

line 328: non-ascii characters found:
- Hans Olsson, Dassault Systèmes, Sweden
                            ^            

line 339: non-ascii characters found:
- Karl Wernersson, Dassault Systèmes, Sweden
                                ^            

line 421: non-ascii characters found:
|_Event_ connected to the tick of an input clock of an FMU that is caused by an external potentially random process, i.e. hardware interrupt, software interrupt.�
                                                                                                                                                                 ^ 

line 460: non-ascii characters found:
|Tool internal time grid for saving output data to file (in some tools also known as "_communication points_" – but this term is used in a different way in FMI for Co-Simulation, see above).
                                                                                                              ^                                                                                

line 515: non-ascii characters found:
|eXtensible Markup Language (http://www.w3.org/XML/[www.w3.org/XML], http://en.wikipedia.org/wiki/Xml[en.wikipedia.org/wiki/XML]) – An open standard to store information in text files in a structured form.
                                                                                                                                  ^                                                                           

Problems found in ./docs/4_1_co-simulation_math.adoc

line 12: non-ascii characters found:
In this framework, the physical connections between subsystems are represented by mathematical coupling conditions between the inputs latexmath:[u(t)] and the outputs latexmath:[y(t)] of all subsystems, _Kübler and Schiehlen (2000)_.
                                                                                                                                                                                                             ^                            

line 195: non-ascii characters found:
Definition (4.1) is consistent with the definition of co-simulation by (Kübler, Schiehlen 2000).
                                                                         ^                       

Problems found in ./docs/5___literature.adoc

line 2: non-ascii characters found:
- Åkesson J., Braun W., Lindholm P., and Bachmann B. (2012): **Generation of Sparse Jacobians for the Functional Mockup Interface 2.0**. 9th International Modelica Conference, Munich, 2012. http://www.ep.liu.se/ecp/076/018/ecp12076018.pdf
   ^                                                                                                                                                                                                                                            

line 6: non-ascii characters found:
- Blochwitz T., Otter M., Arnold M., Bausch C., Clauß C., Elmqvist H., Junghanns A., Mauss J., Monteiro M., Neidhold T., Neumerkel D., Olsson H., Peetz J.-V., Wolf S. (2011): **The Functional Mockup Interface for Tool independent Exchange of Simulation Models**. 8th International Modelica Conference, Dresden 2011. http://www.ep.liu.se/ecp/063/013/ecp11063013.pdf
                                                    ^                                                                                                                                                                                                                                                                                                                        

line 8: non-ascii characters found:
- Blochwitz T., Otter M., Akesson J., Arnold M., Clauß C., Elmqvist H., Friedrich M., Junghanns A., Mauss J,, Neumerkel D., Olsson H., Viel A. (2012): **Functional Mockup Interface 2.0: The Standard for Tool independent Exchange of Simulation Models**. 9th International Modelica Conference, Munich, 2012. http://www.ep.liu.se/ecp/076/017/ecp12076017.pdf
                                                     ^                                                                                                                                                                                                                                                                                                             

line 10: non-ascii characters found:
- Kübler R., Schiehlen, W. (2000): **Two methods of simulator coupling**. Mathematical and Computer Modeling of Dynamical Systems 6 pp. 93-113.
    ^                                                                                                                                            

line 11: non-ascii characters found:
  Lee E.A., Zheng H. (2007): Leveraging Synchronous Language Principles for Heterogeneous Modeling and Design of Embedded Systems. EMSOFT’07, Sept. 30 - Oct. 3, 2007, Salzburg, Austria. https://ptolemy.berkeley.edu/publications/papers/07/unifying/LeeZheng_SRUnifying.pdf
                                                                                                                                         ^                                                                                                                                     

line 13: non-ascii characters found:
- Lee E.A., Zheng H. (2007): **Leveraging Synchronous Language Principles for Heterogeneous Modeling and Design of Embedded Systems**. EMSOFT’07, September 30–October 3, Salzburg, Austria. https://dl.acm.org/citation.cfm?id=1289949
                                                                                                                                             ^                ^                                                                         

Problems found in ./docs/2_2_common_schema.adoc

line 175: non-ascii characters found:
_[Example: copyright = "© My Company 2011"]_.
                        ^                     

line 438: non-ascii characters found:
^|`°F`
   ^   

line 500: non-ascii characters found:
J*α = τ → [kg.m2]*[rad/s2] = [kg.m2/s2]). // o.k. ("rad" is treated as "1")
  ^   ^ ^                                                                   

line 501: non-ascii characters found:
J*α = f → [kg.m2]*[rad/s2] = [kg.m/s2]). // error, since dimensions do not agree
  ^     ^                                                                        

line 532: non-ascii characters found:
_[For example, if latexmath:[\color{blue}{T_K}] is the temperature value of `Unit.name` (in "K") and latexmath:[\color{blue}{T_F}] is the temperature value of `DisplayUnit` (in "°F"),
                                                                                                                                                                                  ^     

line 604: non-ascii characters found:
|Minimum value of variable (variable value ≥ `min`).
                                           ^         

line 609: non-ascii characters found:
|Maximum value of variable (variable value ≤ `max`).
                                           ^         

line 679: non-ascii characters found:
The priority of a clock has to be defined via the integer attribute `priority` – smaller values have a higher priority.
                                                                               ^                                        

line 930: non-ascii characters found:
|Variables which fulfill Kirchhoff’s current law.
                                  ^               

line 958: non-ascii characters found:
_[Flow variables have a direction and must fulfill a zero sum constraint i.e. the sum of all flow variables connected together must be zero (Kirchhoff’s current law)._
                                                                                                                                                      ^                 

line 1131: non-ascii characters found:
Changing any dimension of a variable in *Configuration Mode* or *Reconfiguration Mode* invalidates the variable’s current value (including its start value).
                                                                                                               ^                                             

line 1194: non-ascii characters found:
<Variable name=“spD" valueReference="126" causality=“structuralParameter"
               ^                                    ^                     

line 1195: non-ascii characters found:
  variability=“fixed">
              ^        

line 1315: non-ascii characters found:
*fmi3SetXXX* can be called on any variable with `variability ≠ "constant"` *before initialization* (before calling <<fmi3EnterInitializationMode>>)
                                                             ^                                                                                      

line 1319: non-ascii characters found:
*fmi3SetXXX* can be called on any variable with `variability ≠ "constant"` *during initialization* (after calling <<fmi3EnterInitializationMode>> and before <<fmi3ExitInitializationMode>> is called)
                                                             ^                                                                                                                                         

line 1931: non-ascii characters found:
Introduce a flag “hasInternalState”, see also #148.]
                 ^                ^                  

line 2162: non-ascii characters found:
Q-name          = "’" ( Q-char | escape ) { Q-char | escape } "’"
                   ^                                           ^  

line 2167: non-ascii characters found:
escape          = "\’" | "\"" | "\?" | "\\" | "\a" | "\b" |
                    ^                                       

line 2220: non-ascii characters found:
…
^ 

Problems found in ./docs/2_1_common_api.adoc

line 482: non-ascii characters found:
Variables can be scalar or array variables. When getting or setting the values of array variables, the serialization of array variable values used in C-API function calls, as well as in the XML `start` attributes is defined as row major - i.e. dimension order from left->right for the C-API (e.g. `array[dim1][dim2]…[dimN]`), and document order in the XML attributes.
                                                                                                                                                                                                                                                                                                                           ^                                                    

line 691: non-ascii characters found:
The priority of a clock has to be defined in the modelDescription.xml via the clock variable integer attribute `priority` – smaller values have a higher priority.
                                                                                                                          ^                                        

line 811: non-ascii characters found:
|Don’t call.
    ^        

line 1008: non-ascii characters found:
Efficient graph coloring algorithms are freely available, such as library ColPack (https://cscapes.cs.purdue.edu/coloringpage/) written in C/C++ (LGPL), or the routines by Coleman, Garbow, Moré: "Software for estimating sparse Jacobian matrices", ACM Transactions on Mathematical Software - TOMS, vol. 10, no. 3, pp. 346-347, 1984. See e.g. http://www.netlib.org/toms/618._
                                                                                                                                                                                                 ^                                                                                                                                                                                     

Problems found in ./docs/0___preamble.adoc

line 49: non-ascii characters found:
*Share* — copy and redistribute the material in any medium or format
        ^                                                            

line 51: non-ascii characters found:
*Remix* — remix, transform, and build upon the material for any purpose, even commercially.
        ^                                                                                   

line 57: non-ascii characters found:
*Attribution* — You must give appropriate credit, provide a link to the
              ^                                                         

line 61: non-ascii characters found:
*Share Alike* — If you remix, transform, or build upon the material, you must distribute your contributions under the same license as the original.
              ^                                                                                                                                     

Problems found in ./docs/1___overview.adoc

line 241: non-ascii characters found:
Daimler AG, DLR, ITI GmbH, Martin Luther University Halle-Wittenberg, QTronic GmbH and SIMPACK AG thank BMBF for partial funding of this work within MODELISAR (BMBF Förderkennzeichen: 01lS0800x).
                                                                                                                                                                      ^                             

line 243: non-ascii characters found:
Dassault Systèmes (Sweden) thanks the Swedish funding agency VINNOVA (2008-02291) for partial funding of this work within MODELISAR.
             ^                                                                                                                       

line 249: non-ascii characters found:
- DLR, ITI GmbH, QTronic GmbH and SIMPACK AG thank BMBF for partial funding of this work within MODRIO (BMBF Förderkennzeichen: 01IS12022E).
                                                                                                              ^                              

line 251: non-ascii characters found:
- Dassault Systèmes (Sweden), Linköping University and Modelon AB thank the Swedish funding agency VINNOVA (2012--01157) for partial funding of this work within MODRIO.
               ^                  ^                                                                                                                                      

Problems found in ./docs/4_2_co-simulation_api.adoc

line 964: non-ascii characters found:
The FMU’s code has to be prepared for being able to correctly handle preemptive calls of `fmi3ActivateModelPartition()`,`fmi3Get{VariableType}()`,`fmi3Set{VariableType}()`.
       ^                                                                                                                                                                     

line 976: non-ascii characters found:
Set/get calls for each task are only allowed for variables that are associated to the input clock associated to that task or – here preemption issues become important – to variables that are associated to no clocks (global variables), based on the XML information (see 4.4.3).
                                                                                                                             ^                                         ^                                                                                                             

line 1004: non-ascii characters found:
The preemption lock has to be actively locked and unlocked in the FMU’s code.
                                                                     ^        

Problems found in ./docs/3_1_model_exchange_math.adoc

line 67: non-ascii characters found:
The following variable indices are used to describe the timing behavior of the corresponding variable (for example, 𝑣𝑑 is a discrete-time variable).
                                                                                                                    ^^                               

line 79: whitespace at the end of the line

line 91: whitespace at the end of the line

line 380: non-ascii characters found:
_At an event instant, the discrete system is described by algebraic equations as function of the previous (internal) discrete-time states_ latexmath:[_{‍}^{\bullet}\mathbf{x}_{d}] _and the discrete-time inputs_ latexmath:[\mathbf{u}_{d}]__.
                                                                                                                                                        ^                                                                                        

line 382: non-ascii characters found:
If the actual discrete-time states__ latexmath:[\mathbf{x}_{d}] __and the previous discrete-time states__ latexmath:[_{‍}^{\bullet}\mathbf{x}_{d}] _are not identical, the discrete-time states are updated, the Integer part of the time is incremented and a new event iteration is performed.
                                                                                                                       ^                                                                                                                                                                         

Problems found in ./docs/4___co-simulation.adoc

line 0: non-ascii characters found:
== FMI for Co‑Simulation [[fmi-for-co-simulation]]
             ^                                     

Total problems found: 84
```

**Comment by chrbertsch** [comment]
Was fixed in PR #689
